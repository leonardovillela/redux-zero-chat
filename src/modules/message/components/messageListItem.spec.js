import React from 'react';
import { shallow } from 'enzyme';
import Chance from 'chance';

describe('Test MessageListItem component unit', () => {
  const chance = new Chance();

  beforeEach(() => {
    jest.resetModules();
  });

  it('Check if dateFormatted is called', () => {
    const dateStub = chance.string();
    const formattedDateStub = chance.string();
    const dateFormatterMock = jest.fn(() => formattedDateStub);

    jest.doMock('../../../utils', () => ({ dateFormatter: { format: dateFormatterMock } }));
    const MessageListItem = require('./messageListItem').default;

    const wrapper = shallow(
      <MessageListItem
        createdAt={dateStub}
        content=""
        userName=""
      />
    );

    expect(dateFormatterMock).toBeCalledWith(dateStub);
    expect(wrapper.find('small').text()).toEqual(formattedDateStub);
  });

  it('Should render component with correct structure', () => {
    jest.doMock('../../../utils', () => ({ dateFormatter: { format: () => 'Date formatted' } }));
    const MessageListItem = require('./messageListItem').default;

    const wrapper = shallow(<MessageListItem createdAt="" userName="Test username" content="Test Content" />);
    expect(wrapper).toMatchSnapshot();
  })
});
