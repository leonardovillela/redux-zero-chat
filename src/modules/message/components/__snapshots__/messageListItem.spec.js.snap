// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MessageListItem component unit Should render component with correct structure 1`] = `
<div
  className="w-100 bb pv1"
>
  <h2
    className="di v-mid ma0"
  >
    Test username
  </h2>
  <small
    className="di fr v-mid"
  >
    Date formatted
  </small>
  <p>
    Test Content
  </p>
</div>
`;
